src/GraphQL.php

```diff
@@ -94,11 +73,37 @@ public function schema($schema = null): Schema
             'name' => 'Subscription',
         ]);
 
+
+
+
         return new Schema([
-            'query'         => $query,
-            'mutation'      => ! empty($schemaMutation) ? $mutation : null,
-            'subscription'  => ! empty($schemaSubscription) ? $subscription : null,
-            'types'         => $types,
+            'query' => $query,
+            'mutation' => !empty($schemaMutation) ? $mutation : null,
+            'subscription' => !empty($schemaSubscription) ? $subscription : null,
+            'typeLoader' => function ($name) {
+                $keyName = array_search($name, $this->typesInstances);
```

Comment:
We should try to avoid `array_search` as it has linear performance characteristics.

Reply:
> We should try to avoid `array_search` as it has linear performance characteristics.

waiting to change this, to after I have done some benchmarking

Reply:
Do you have any results?